tools/pythonpkg/tests/fast/spark/test_spark_dataframe.py

```diff
@@ -47,22 +58,29 @@ def test_dataframe_from_list_of_tuples(self, spark):
             df = spark.createDataFrame(5, ["id", "address", "test"])
 
         # Empty list
-        df = spark.createDataFrame([], ["id", "address", "test"])
-        res = df.collect()
-        assert res == []
+        if not USE_ACTUAL_SPARK:
+            # Spark raises PySparkValueError [CANNOT_INFER_EMPTY_SCHEMA]
+            df = spark.createDataFrame([], ["id", "address", "test"])
+            res = df.collect()
+            assert res == []
 
         # Duplicate column names
         address = [(1, "14851 Jeffrey Rd", "DE"), (2, "43421 Margarita St", "NY"), (3, "13111 Siemon Ave", "DE")]
         df = spark.createDataFrame(address, ["id", "address", "id"])
         res = df.collect()
-        assert (
-            str(res)
-            == "[Row(id=1, address='14851 Jeffrey Rd', id='DE'), Row(id=2, address='43421 Margarita St', id='NY'), Row(id=3, address='13111 Siemon Ave', id='DE')]"
-        )
+        exptected_res_str = "[Row(id=1, address='14851 Jeffrey Rd', id='DE'), Row(id=2, address='43421 Margarita St', id='NY'), Row(id=3, address='13111 Siemon Ave', id='DE')]"
+        if USE_ACTUAL_SPARK:
+            # Spark uses string for both ID columns. DuckDB correctly infers the types.
+            exptected_res_str = (
+                exptected_res_str.replace("id=1", "id='1'").replace("id=2", "id='2'").replace("id=3", "id='3'")
+            )
+        assert str(res) == exptected_res_str
 
         # Not enough column names
-        with pytest.raises(PySparkValueError, match="number of columns in the DataFrame don't match"):
-            df = spark.createDataFrame(address, ["id", "address"])
+        if not USE_ACTUAL_SPARK:
+            # Spark does not raise this error
```

Comment:
`FIXME: `
Odd behavior actually, I guess they assign some default column name to the third one? Something I'll have to look into at some point 👍